auth/src/main/java/com/firebase/ui/auth/provider/TwitterProvider.java

```diff
@@ -60,8 +74,36 @@ public void startLogin(Activity activity) {
     }
 
     @Override
-    public void success(Result<TwitterSession> result) {
-        mCallbackObject.onSuccess(createIdpResponse(result.data));
+    public void success(final Result<TwitterSession> sessionResult) {
```

Comment:
@samtstern Whoa, you just blew my mind! The main reason I didn't want to use `requestEmail` was because of the leak, but now that you showed me how to use `WeakReference`s... Wow! This is why I love contributing to FirebaseUI so much: I've read a few Medium posts on weak references, but I need people like you to show me where I can apply the concepts. Anyway, thanks for the tip! (I had to go back and rewrite some code in my app to use weak references 😄)